src/objects.c

```diff
@@ -425,13 +425,14 @@ void PrepareCopy(Obj obj, Obj copy)
     // between original and copy (e.g. for objects, they point to the type;
     // if making an immutable copy of a mutable object, the types will
     // differ).
+    // Store in a DATOBJ, as CONST_ADDR_OBJ(obj)[0] could be anything,
+    // including an arbitary C integer.
     // Likewise, the TNUM of the copy and the original can and will differ;
     // e.g. for a weak pointer list, the copy can be a plist.
-    Obj tmp = NEW_PLIST(T_PLIST, 3);
-    SET_LEN_PLIST(tmp, 3);
-    SET_ELM_PLIST(tmp, 1, CONST_ADDR_OBJ(obj)[0]);
-    SET_ELM_PLIST(tmp, 2, copy);
-    SET_ELM_PLIST(tmp, 3, INTOBJ_INT(TNUM_OBJ(obj)));
+    Obj tmp = NewBag(T_DATOBJ, 4 * sizeof(Obj *));
+    ADDR_OBJ(tmp)[1] = CONST_ADDR_OBJ(obj)[0];
+    ADDR_OBJ(tmp)[2] = copy;
```

Comment:
But this is the only reference to `copy` and GASMAN cannot see it here. So unless I miss something, this makes it worse...

Reply:
I actually have a different way of handling this, which might be better / safer.

Reply:
So the problem isn't GASMAN, the problem is SaveWorkspace, which walks along the plist and expects to find a valid GAP Obj at every location. GASMAN is actually happy with getting random pointers (as it has to be, as they might be found on the stack).

Reply:
Of course GASMAN is fine with this in the current form in master. But I don't think it is OK with your T_DATOBJ variant, as no subbags / slots other than that at offset 0 are marked.

Reply:
I added this to MarkCopyingSubbags, which already special-cases the things it copies.

Reply:
But  how does the work? If the only reference to `copy` is in that `T_DATOBJ`, then on a full GC, we never mark `copy`, and hence never invoke `MarkCopyingSubbags` on it -- or am I missing something?